src/screens/AddSiteScreen.js

```diff
@@ -125,24 +142,45 @@ class AddSiteScreen extends Component {
         this.props.logUserIntoFacebook({navigate});
     };
 
+    renderWaitingOnConnectionIfNecessary() {
+        const {pendingUploadSites} = this.props;
+        const {infoTextStyle, errorTextStyle, waitingSignalRowStyle, leftMargin} = styles;
+
+        const thereAreSitesToUpload = this.thereAreSitesToUpload({pendingUploadSites});
+        const connectionIsStrongEnough = this.connectionIsStrongEnough(this.props);
+
+        if (thereAreSitesToUpload && !connectionIsStrongEnough) {
+            return (
+                <View style={waitingSignalRowStyle}>
+                    <ActivityIndicator size='large'/>
+                    <Text style={[infoTextStyle, errorTextStyle, leftMargin]}>{pending_upload_sites_waiting}</Text>
+                </View>
+            );
+        } else {
+            return null;
+        }
```

Comment:
Now provide a nice helper text with spinner if we're waiting on a better connection before auto-upload.